### PR TITLE
Getting hands on on a microprofile project

### LINKED ISSUE #2978: Getting hands-on on a MicroProfile project
### Proposed title of the article

Getting hands-on on a MicroProfile project

### Introduction paragraph (2-3 paragraphs)

MicroProfile specifies a collection of Java EE (Enterprise) technologies and APIs which in consolidation from a core baseline microservice that aims to deliver application portability across multiple MicroProfile runtimes.
It has been attracting attention from developers since this Java EE implementation does not take up much time in server booting and code run time as it did before when using former versions of Java EE. This is due to the continuous agile improvement of individual Java Specification Requests (JARs) that make up Java Enterprise (EE).

At the edge of this course, we will learn how MicroProfiles work and how they are used and how to use them. We shall also get hands-on one on MicroProfile Projects.

### Key takeaways

Coming to the tip of the article, we shall have the following to take with us:

- What is MicroProfile
- MicroProfile prerequisites
- Advantages and trade-offs of MicroProfile framework
- Creation of a MicroProfile application using different methods or tools such as from the [https://start.microprofile.io/](https://start.microprofile.io/) website, using Visual Studio Code and IntelliJ.
- MicroProfile project structure
- MicroProfile vs Spring Boot

### Pre-requisites

Some basics required for easy follow-up on this article include:

- A modern IDE that supports Java frameworks installed on your machine. This may either be Visual Studio Code or IntelliJ or any other. We shall focus on the two in this article.
- A stable internet connection for dependency downloads.
- **JDK 11+** of any distribution installed. This is because MicroProfile is not supported with lower versions than 11.
- Gradle or Apache Maven 3.8.1+ installed.
- Java programming skills (It will be a bit hard for a beginner to understand or catch up)

### References

[microprofile-3-0-support-comes-to-helidon](https://medium.com/helidon/microprofile-3-0-support-comes-to-helidon-f72ba752b42f)

### Templates to use as guides

- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

### PR DESCRIPTION
**Pull request by justusmbuvi**
### Article checklist

**ATTENTION: In your PR - add the ISSUE (topic form) # that this PR corresponds to. You can do this by also adding a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.**
      
      
**NOTE: (Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**

**If you have not already please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve our overall technical writing.**

#### Preliminary Checks - Formatting and Structure 
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker? We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**

This fixes #2978

### CODE REVIEW

**Comment by justusmbuvi** [comment]
This fixes #2978

**Comment by okeloviolet** [comment]
@hectorkambow I'll be reviewing this article.

**Comment by justusmbuvi** [comment]
Will be waiting for @okeloviolet and @paulodhiambo for the pings...

**Comment by hectorkambow** [comment]
plagiarism check done

File: content/articles/getting-hands-on-on-a-microprofile-project/index.md [hunk 1]
```diff
@@ -0,0 +1,416 @@
+MicroProfile specifies a collection of Java EE (Enterprise) technologies and APIs which in consolidation from a core baseline microservice that aims to deliver application portability across multiple MicroProfile runtimes.
```
Comment (okeloviolet) [inline content/articles/getting-hands-on-on-a-microprofile-project/index.md:1]: Change "...which in consolidation..." to "..that consolidate..."

**Comment by okeloviolet** [comment]
@justusmbuvi amazing tutorial. I am performing a technical reviewing.

**Comment by justusmbuvi** [comment]
> @justusmbuvi amazing tutorial. I am performing a technical reviewing.

Will be waiting

File: content/articles/getting-hands-on-on-a-microprofile-project/index.md (hunk 1 above)
Comment (justusmbuvi) [reply content/articles/getting-hands-on-on-a-microprofile-project/index.md:1]: Ok

Comment (justusmbuvi) [reply content/articles/getting-hands-on-on-a-microprofile-project/index.md]: Done! 🚀

**Comment by okeloviolet** [comment]
@justusmbuvi amazing article.
ready for final review

**Comment by hectorkambow** [comment]
Overall nice job.  @justusmbuvi 

Please do go over my final edits - as I made many changes before publishing. (some for format - some for flow.)

Please review to improve future articles and reviews - as articles with too many edits/revisions will be closed. 

👍  

cc @okeloviolet

**Pull request merged**